test-tools/core/src/main/java/org/kie/cloud/tests/core/resources/Project.java

```diff
@@ -0,0 +1,329 @@
+                    .timeout(TimeUnit.MINUTES, 5)
+                    .waitFor();
+        }
+        return new Project(projectName);
+    }
+
+    /**
+     * @return Project name.
+     */
+    public String getName() {
+        return projectName;
+    }
+
+    /**
+     * @return OpenShift client.
+     */
+    public OpenShift getOpenShift() {
+        return openShift;
+    }
+
+    /**
+     * @return OpenShift admin client.
+     */
+    public OpenShift getOpenShiftAdmin() {
+        return openShiftAdmin;
+    }
+
+    /**
+     * Delete OpenShift project.
+     */
+    public void delete() {
+        openShift.deleteProject();
+    }
+
+    /**
+     * Process template and create all resources defined there.
+     *
+     * @param templateUrl URL of template to be processed
+     * @param envVariables Map of environment variables to override default values from the template
+     */
+    public void processTemplateAndCreateResources(URL templateUrl, Map<String, String> envVariables) {
+        boolean templateIsFile = templateUrl.getProtocol().equals("file");
+
+        // Used to log into OpenShift
+        OpenShiftBinary oc = openShiftBinaryClient();
+
+        List<String> commandParameters = new ArrayList<>();
+        commandParameters.add(openShiftBinaryPath());
+        commandParameters.add("process");
+        commandParameters.add("-f");
+        commandParameters.add(templateIsFile ? templateUrl.getPath() : templateUrl.toExternalForm());
+        commandParameters.add("--local");
+        commandParameters.add("--ignore-unknown-parameters=true");
+        commandParameters.add("-o");
+        commandParameters.add("yaml");
+        for (Entry<String, String> envVariable : envVariables.entrySet()) {
+            commandParameters.add("-p");
+            commandParameters.add(envVariable.getKey() + "=" + envVariable.getValue());
+        }
+        String completeProcessingCommand = commandParameters.stream().collect(Collectors.joining(" "));
+
+        try (ProcessExecutor executor = new ProcessExecutor()) {
+            File processedTemplate = executor.executeProcessCommandToTempFile(completeProcessingCommand);
+            oc.execute("create", "-n", getName(), "-f", processedTemplate.getAbsolutePath());
+        }
+        // TODO: Temporary workaround to wait until scenario is completely initialized as there is a delay between finishing template creation command
+        // and actual creation of resources on OpenShift. This should be removed when deployments won't be scaled in the beginning and will contain availability check.
+        try {
+            TimeUnit.SECONDS.sleep(10L);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Interrupted while waiting for scenario to be initialized.", e);
+        }
+    }
+
+    static Semaphore semaphore = new Semaphore(1);
```

Comment:
Semaphore has been brought by @jakubschwan , I think that we had a discussion that this could be replaced by synchronization blocks with static lock (by project class ?). Semaphore is from my point a bit of overkill for our purposes - synchronizing access to oc client.
Maybe OC client usage can separated into dedicated method accepting consumer and being synchronized there.

Reply:
This was bring for APB image run as this in parallel were projects switched and run command failed. During implementation were also projects switched when was used synchronization block.

Reply:
Was the block statically synchronized?

Reply:
as I remember, yes

Reply:
hmm, maybe that could be compensated by adding namespace parameter for every command